src/controllers/auth-controller.ts

```diff
@@ -10,22 +13,97 @@ export default class AuthController extends AbstractController {
 
   private users: Repository<User>;
 
+  private tokens: Repository<RefreshToken>;
+
   public init(): Router {
     this.router = Router();
     this.users = getRepository(User);
+    this.tokens = getRepository(RefreshToken);
+    
     // @ts-ignore
     this.router.post('/signup', validateSchema(Registration), this.signUp);
+    this.router.post('/logout', this.logout);
+    this.router.post('/login', this.login);
+    this.router.post('/refresh', this.refresh);
 
+    /* use auth.required to secure route */
+    this.router.get('/test', auth.required, this.test)
     return this.router;
   }
 
   private signUp = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
       const user = await User.create(req.body);
       await this.users.save(user);
-      res.json(user);
+      const { token, refreshToken } = singTokens({ userId: user.id, userRole: user.role });
+      await this.tokens.save(new RefreshToken(refreshToken));
+      res.json({ user, token: `Bearer ${token}`, refreshToken });
     } catch (e) {
       next(e);
     }
   };
+
+  private logout = async (req: Request, res: Response, next: NextFunction) => {
+    const { refreshToken }: { refreshToken: string } = req.body;
+    if (refreshToken) {
+      try {
+        const token = await this.tokens.findOne({ token: refreshToken });
+        if (token) {
+          await this.tokens.delete(token.id);
+          res.status(200).end();
```

Comment:
one note 200 is default)
`res.end()` is enough

But it's more up to used convention between server and client
F.e. they can wait for
`res.send({ok: true})` or `res.send({success: true, removed: true})`